src/tensorwaves/model/sympy.py

```diff
@@ -414,16 +406,22 @@ def __init__(
         )
         self.max_complexity = max_complexity
 
-    def lambdify(self, backend: Union[str, tuple, dict]) -> Callable:
+    def lambdify(self, backend: Union[str, tuple, dict]) -> Function:
         """Lambdify the model using `~sympy.utilities.lambdify.lambdify`."""
-        return _sympy_lambdify(
+        if self.max_complexity is None:
```

Comment:
Why is this public btw? I could not find any doc string about this or quick reasoning to make sense of it

Reply:
Do you mean the `lambdify` method?

Reply:
Actually, while working on this PR, I started to have some doubts about the implementation classes. There is a lot of logics going on in the constructor. Would it be better to create some functions that build such classes? In this case, a `lambdify` **function** (could be a static method) that builds this `ParametrizedBackendFunction` from a SymPy expression? Currently, there is only one way customise the construction process, through the constructor.

Reply:
Ah sorry, I was talking about the `max_complexity`

Reply:
Fixed by #357